zone/spell_effects.cpp

```diff
@@ -7109,331 +7111,948 @@ bool Mob::ImprovedTaunt(){
+				return true;
+			break;
 
-					if (value == 400 && GetHPRatio() <= 25)
-							return true;
+		case IS_CLASS_CLR_SHM_DRU:
+			if (IsWISCasterClass(GetClass()))
+				return true;
+			break;
 
-					else if (value == base2_value){
-						if (GetHPRatio() > 25+((base2_value - 401)*10) && GetHPRatio() <= 35+((base2_value - 401)*10))
-							return true;
-					}
-				}
+		case IS_CLASS_NOT_WAR_PAL_SK:
+			if ((GetClass() != WARRIOR) && (GetClass() != PALADIN) && (GetClass() != SHADOWKNIGHT))
+				return true;
+			break;
+
+		case IS_LEVEL_UNDER_100:
+			if (GetLevel() < 100)
+				return true;
+			break;
+
+		case IS_NOT_RAID_BOSS:
+			if (!IsRaidTarget())
+				return true;
+			break;
+
+		case IS_RAID_BOSS:
+			if (IsRaidTarget())
+				return true;
+			break;
+
+		case FRENZIED_BURNOUT_ACTIVE:
+			if (HasBuffWithSpellGroup(SPELLGROUP_FRENZIED_BURNOUT))
+				return true;
+			break;
+
+		case FRENZIED_BURNOUT_NOT_ACTIVE: 
+			if (!HasBuffWithSpellGroup(SPELLGROUP_FRENZIED_BURNOUT))
+				return true;
+			break;
+
+		case IS_HP_ABOVE_75_PCT:
+			if (GetHPRatio() > 75)
+				return true;
+			break;
+			
+		case IS_HP_LESS_THAN_20_PCT:
+			if (GetHPRatio() <= 20)
+				return true;
+			break;
+
+		case IS_HP_LESS_THAN_50_PCT:
+			if (GetHPRatio() <= 50)
+				return true;
+			break;
+
+		case IS_HP_LESS_THAN_75_PCT:
+			if (GetHPRatio() <= 75)
+				return true;
+			break;
+
+		case IS_NOT_IN_COMBAT:
+			if (!IsEngaged())
+				return true;
+			break;
+
+		case IS_HP_LESS_THAN_35_PCT:
+			if (GetHPRatio() <= 35)
+				return true;
+			break;
+
+		case HAS_BETWEEN_1_TO_2_PETS_ON_HATELIST: {
+			int count = hate_list.GetSummonedPetCountOnHateList(this);
+			if (count >= 1 && count <= 2) {
+				return true;
 			}
+			break;
 		}
 
-		else if (value >= 500 && value <= 549){
-			for (int base2_value = 500; base2_value <= 520; ++base2_value){
-				if (value == base2_value){
-					if (GetHPRatio() < (base2_value - 500)*5)
-						return true;
-				}
+		case HAS_BETWEEN_3_TO_5_PETS_ON_HATELIST: {
+			int count = hate_list.GetSummonedPetCountOnHateList(this);
+			if (count >= 3 && count <= 5) {
+				return true;
+			}
+			break;
+		}
+
+		case HAS_BETWEEN_6_TO_9_PETS_ON_HATELIST: {
+			int count = hate_list.GetSummonedPetCountOnHateList(this);
+			if (count >= 6 && count <= 9) {
+				return true;
+			}
+			break;
+		}
+
+		case HAS_BETWEEN_10_TO_14_PETS_ON_HATELIST: {
+			int count = hate_list.GetSummonedPetCountOnHateList(this);
+			if (count >= 10 && count <= 14) {
+				return true;
 			}
+			break;
 		}
 
-		else if (value == 399) {
-			if (GetHPRatio() > 15 && GetHPRatio() <= 25)
+		case HAS_MORE_THAN_14_PETS_ON_HATELIST: {
+			int count = hate_list.GetSummonedPetCountOnHateList(this);
+			if (count > 14) {
 				return true;
+			}
+			break;
 		}
-	} // End Heal
 
+		case IS_CLASS_CHAIN_OR_PLATE:
+			if (IsClient() &&
+				((GetClass() == WARRIOR) || (GetClass() == BARD)  || (GetClass() == SHADOWKNIGHT)  || (GetClass() == PALADIN)  || (GetClass() == CLERIC)
```

Comment:
This could have been summarized with helpers `IsChain(GetClass())` or `IsPlate(GetClass())`